modules/home-age.nix

```diff
@@ -0,0 +1,135 @@
+
+  installSecret = secretType: ''
+    _truePath="${secretType.path}"
+    echo "decrypting '${secretType.file}' to '$_truePath'..."
+    TMP_FILE="$_truePath.tmp"
+    mkdir -p "$(dirname "$_truePath")"
+    [ "${secretType.path}" != "${cfg.secretsDir}/${secretType.name}" ] && mkdir -p "$(dirname "${secretType.path}")"
+    (
+      umask u=r,g=,o=
+      test -f "${secretType.file}" || echo '[agenix] WARNING: encrypted file ${secretType.file} does not exist!'
+      test -d "$(dirname "$TMP_FILE")" || echo "[agenix] WARNING: $(dirname "$TMP_FILE") does not exist!"
+      ${if config.home.language.base != null then "LANG=${config.home.language.base}" else ""} ${ageBin} --decrypt ${identities} -o "$TMP_FILE" "${secretType.file}"
+    )
+    chmod ${secretType.mode} "$TMP_FILE"
+    chown ${config.home.username}:${secretType.group} "$TMP_FILE"
+    mv -f "$TMP_FILE" "$_truePath"
+    ${optionalString secretType.symlink ''
+      [ "${secretType.path}" != "${cfg.secretsDir}/${secretType.name}" ] && ln -sfn "${cfg.secretsDir}/${secretType.name}" "${secretType.path}"
+    ''}
+  '';
+
+  testIdentities = map
+    (path: ''
+      test -f ${path} || echo '[agenix] WARNING: config.age.identityPaths entry ${path} not present!'
+    '')
+    cfg.identityPaths;
+
+  installSecrets = builtins.concatStringsSep "\n" ([ "echo '[agenix] decrypting user secrets...'" ] ++ (map installSecret (builtins.attrValues cfg.secrets)));
+
+  secretType = types.submodule ({ config, ... }: {
+    options = {
+      name = mkOption {
+        type = types.str;
+        default = config._module.args.name;
+        description = ''
+          Name of the file used in ''${cfg.secretsDir}
+        '';
+      };
+      file = mkOption {
+        type = types.path;
+        description = ''
+          Age file the secret is loaded from.
+        '';
+      };
+      path = mkOption {
+        type = types.str;
+        default = "${cfg.secretsDir}/${config.name}";
+        description = ''
+          Path where the decrypted secret is installed.
+        '';
+      };
+      mode = mkOption {
+        type = types.str;
+        default = "0400";
+        description = ''
+          Permissions mode of the decrypted secret in a format understood by chmod.
+        '';
+      };
+      # Don't know how to make it find config.home.username becayse this "age" module is a "submodule"
+      #  owner = mkOption {
+      #    type = types.str;
+      #    readOnly = true;
+      #    default = "${config.home.username}";
+      #    description = ''
+      #      User of the decrypted secret.
+      #    '';
+      #  };
+      group = mkOption {
+        type = types.str;
+        default = "$(id -gn)";
+        description = ''
+          Group of the decrypted secret.
+        '';
+      };
+      symlink = mkEnableOption "symlinking secrets to their destination" // { default = true; };
+    };
+  });
+in
+{
+  options.age = {
+    ageBin = mkOption {
```

Comment:
What do you think about `package` instead of `ageBin`?

Reply:
Sounds good, will change tomorrow